CRAMv3.tex

```diff
@@ -920,138 +921,325 @@ \section{\textbf{Record structure}}
+Note however some of these flags can be derived during decode, so may be omitted in the CRAM file and the bits computed based on both reads of a pair-end library residing within the same slice.
+
+\begin{tabular}{|>{\raggedright}p{144pt}|>{\raggedright}p{144pt}|>{\raggedright}p{144pt}|}
+\hline
+\textbf{Bit flag} & \textbf{Comment} & \textbf{Description}\tabularnewline
+\hline
+0x1 & ! 0x40 \&\& ! 0x80 & template having multiple segments in sequencing\tabularnewline
 \hline
-8 & byte[ ] & RN & read name & the read names (if preserved)\tabularnewline
+0x2 &  & each segment properly aligned according to the aligner\tabularnewline
 \hline
-9 & *2 & *2 & mate record & *2 (if not the last record)\tabularnewline
+0x4 &  & segment unmapped\tabularnewline
 \hline
-10 & int & TL & tag ids & tag ids *3\tabularnewline
+0x8 & calculated* or stored in the mate's info & next segment in temlate unmapped\tabularnewline
 \hline
-11 & byte[ ] & - & tag values & tag values *3\tabularnewline
+0x10 &  & SEQ being reverse complemented\tabularnewline
+\hline
+0x20 & calculated* or stored in the mate's info & SEQ of the next segment in the
+template being reversed\tabularnewline
+\hline
+0x40 &  & the first segment in the template\tabularnewline
+\hline
+0x80 &  & the last segment in the template\tabularnewline
+\hline
+0x100 &  & secondary alignment\tabularnewline
+\hline
+0x200 &  & not passing quality controls\tabularnewline
+\hline
+0x400 &  & PCR or optical duplicate\tabularnewline
+\hline
+0x800 &  & Supplementary alignment\tabularnewline
+\hline
+\end{tabular}
+
+* For segments within the same slice.
+
+\subsubsection*{\textbf{CRAM bit flags (CF data series)}}
+
+The CRAM bit flags (also known as compression bit flags) expressed as an integer represent the CF data series. 
+The following compression flags are defined for each CRAM read record:
+
+\begin{tabular}{|>{\raggedright}p{39pt}|>{\raggedright}p{150pt}|>{\raggedright}p{242pt}|}
+\hline
+\textbf{Bit flag} & \textbf{Name} & \textbf{Description}\tabularnewline
+\hline
+0x1 & quality scores stored as array & quality scores can be stored as read features
+or as an array similar to read bases.\tabularnewline
+\hline
+0x2 & detached & mate information is stored verbatim (e.g. because the pair spans multiple slices or the fields differ to the CRAM computed method)\tabularnewline
+\hline
+0x4 & has mate downstream & tells if the next segment should be expected further
+in the stream\tabularnewline
+\hline
+0x8 & decode sequence as ``*'' & informs the decoder that the sequence
+is unknown and that any encoded reference differences are present only to
+recreate the CIGAR string.\tabularnewline
 \hline
 \end{tabular}
 
-*1 The AP data series is delta encoded for reads mapped to a single reference slice 
-and normal integer value in all other cases. 
 
-*2 See \emph{mate record} section.
+The following pseudocode describes the general process of decoding an entire CRAM record.
```

Comment:
This is excellent, thank you.

Reply:
Thanks. :)

I am aware the codec section could be rewritten using algorithmicx too for consistency, but I added it after having gone through them.  I think this is maybe a second round of updates.  There's a lot more to do and incremental improvements is the easiest way given how much of the spec still needs a refresh.